tests/background_cpu_test.py

```diff
@@ -61,24 +67,46 @@ def _CreateBenchmarkSpec(self, benchmark_config_yaml):
         NAME, flag_values=self._mocked_flags, **config)
     return benchmark_spec.BenchmarkSpec(config_spec, NAME, UID)
 
-  def _CheckVMFromSpec(self, spec, num_working):
-    vm0 = spec.vms[0]
-    with patch.object(
-        vm0.__class__, 'RemoteCommand'), patch.object(
-            vm0.__class__, 'Install'):
+  def _CheckVmCallCounts(self, spec, working_groups, working_expected_counts,
+                         non_working_groups, non_working_expected_counts):
+    # TODO(skschneider): This is also used in TestBackgroundNetworkWorkload.
+    # Consider moving to a shared function or base class.
+    expected_call_counts = {group: working_expected_counts
+                            for group in working_groups}
+    expected_call_counts.update({group: non_working_expected_counts
+                                 for group in non_working_groups})
+    for group_name, vm_expected_call_counts in expected_call_counts.iteritems():
+      group_vms = spec.vm_groups[group_name]
+      self.assertEqual(len(group_vms), 1,
+                       msg='VM group "{0}" had {1} VMs'.format(group_name,
+                                                               len(group_vms)))
+      vm = group_vms[0]
+      iter_mocked_functions = itertools.izip_longest(_MOCKED_VM_FUNCTIONS,
+                                                     vm_expected_call_counts)
+      for function_name, expected_call_count in iter_mocked_functions:
+        call_count = getattr(vm, function_name).call_count
+        self.assertEqual(call_count, expected_call_count, msg=(
+            'Expected {0} from VM group "{1}" to be called {2} times, but it '
+            'was called {3} times.'.format(function_name, group_name,
+                                           expected_call_count, call_count)))
+
+  def _CheckVMFromSpec(self, spec, working_groups=(), non_working_groups=()):
+    with contextlib2.ExitStack() as stack:
```

Comment:
ExitStack is a standard python 3 function: https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack
contextlib2 is a python package that backports it to python 2.